Sources/Datadog/DatadogInternal/MessageBus/FeatureMessage.swift

```diff
@@ -0,0 +1,30 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+
+/// The set of messages that can be transimtted on the Features message bus.
+public enum FeatureMessage {
+    /// An error message.
+    case error(
+        message: String,
+        attributes: [String: Encodable]?
+    )
+
+    /// An encodable event that will be transmitted
+    /// as-is through a Feature.
+    case event(
+        type: String,
+        encodable: Encodable
+    )
+
+    /// A custom message with generic encodable
+    /// attributes.
+    case custom(
+        key: String,
+        attributes: [String: Encodable]?
+    )
+}
```

Comment:
@ncreated I'm adding this constraint on the message that can go through the bus. It adds a bit of convenience and safety when processing a message. LMKWYT!

Reply:
Good idea 👍. I like the classification of error / event / custom 👌. I only wonder if modeling this as an `enum` will be convenient on both sides. It will be easy for sender, but also quite constraining for receiver, as it would need to centralise handling. 

How about using protocol + structs as we do with [`RUMCommand`](https://github.com/DataDog/dd-sdk-ios/blob/develop/Sources/Datadog/RUM/RUMMonitor/RUMCommand.swift)? That would IMO promote type safety further, as parts of the receiver would only depend on concrete messages (e.g. `ErrorMessage`), not the entire `FeatureMessage`.

Reply:
The main purpose of using an `enum` is to restrict the type of messages that can go through and to enforce usage of attributes as dictionary. By using a protocol, we would allow extension and open the door to type sharing.
That being said, if we need to add to many cases to the enum I will probably turn this into a protocol. I will see while migrating feature integration 👍

Reply:
Indeed, having public interface would require checking against unknown implementations 👍. Let's go with `enum` and see how it works in handling code 👌.